bitsandbytes/functional.py

```diff
@@ -1197,68 +1176,72 @@ def quantize_4bit(
 
     assert blocksize in [4096, 2048, 1024, 512, 256, 128, 64]
 
-    prev_device = pre_call(A.device)
     is_on_gpu([A, out, absmax])
     if A.dtype == torch.float32:
         if quant_type == "fp4":
-            lib.cquantize_blockwise_fp32_fp4(
-                get_ptr(None),
-                get_ptr(A),
-                get_ptr(absmax),
-                get_ptr(out),
-                ct.c_int32(blocksize),
-                ct.c_int(n),
-            )
+            with torch.cuda.device_of(A):
```

Comment:
This looks like the `with torch.cuda.device_of(A):` block should be wrapping the whole thing, not each invocation of the function.

Looks like this could also use the mapping approach championed above:

```
fn_map = {
  (torch.float32, "fp4"): "cquantize_blockwise_fp32_fp4",
  ...
}
```

Reply:
Simplified this.